runtime/src/accounts_db.rs

```diff
@@ -9948,17 +9994,35 @@ pub mod tests {
         let entry2 = Arc::new(AccountStorageEntry::new(
             dummy_path, dummy_slot, dummy_id2, dummy_size,
         ));
-        entry2.alive_bytes.store(3000, Ordering::Relaxed);
+        entry2.alive_bytes.store(0, Ordering::Relaxed);
         candidates
             .entry(dummy_slot)
             .or_default()
             .insert(entry2.append_vec_id(), entry2.clone());
 
-        let output_candidates =
-            accounts.select_candidates_by_total_usage(&candidates, DEFAULT_ACCOUNTS_SHRINK_RATIO);
+        let dummy_id3 = 55;
+        let entry3 = Arc::new(AccountStorageEntry::new(
+            dummy_path, dummy_slot, dummy_id3, dummy_size,
+        ));
+        entry3.alive_bytes.store(1000, Ordering::Relaxed);
+        candidates
+            .entry(dummy_slot)
+            .or_default()
+            .insert(entry3.append_vec_id(), entry3.clone());
+
+        let (output_candidates, next_candidates) =
+            accounts.select_candidates_by_total_usage(&candidates, 0.17);
         assert_eq!(1, output_candidates.len());
         assert_eq!(1, output_candidates[&dummy_slot].len());
-        assert!(output_candidates[&dummy_slot].contains(&entry2.append_vec_id()));
+        assert!(
+            output_candidates[&dummy_slot].contains(&entry1.append_vec_id())
+                || output_candidates[&dummy_slot].contains(&entry2.append_vec_id())
+        );
+        assert_eq!(1, next_candidates.len());
+        assert!(
+            next_candidates[&dummy_slot].contains(&entry1.append_vec_id())
+                || next_candidates[&dummy_slot].contains(&entry2.append_vec_id())
+        );
```

Comment:
Hmmm why is this not deterministic between `entry1` and `entry2`, is it because their alive ratios are the same?

Could we make this more deterministic by making one of them have a greater alive ratio

Reply:
Will address